transport/udp/udp.go

```diff
@@ -0,0 +1,390 @@
+// Copyright 2016 The Mangos Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use file except in compliance with the License.
+// You may obtain a copy of the license at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package udp implements the UDP transport for mangos. It is based on
+// M. Sustrik, Ed., "UDP Mapping for Scalability Protocols",
+// sp-udp-mapping-01 (March 2014).
+package udp
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/go-mangos/mangos"
+)
+
+// Errors produced by this transport prototocol
+var (
+	ErrMsgTruncated = errors.New("message truncated due to insufficient buffer space")
+)
+
+// connHeader is exchanged during the initial handshake.
+// FIXME: stolen from conn.go -- unify this
```

Comment:
Agreed, just use conn.go's header.  But lets make sure that the headers are the same.  Alternatively, this could be "udpHeader", which happens to be the same but lets the two evolve separately.